src/main/java/core/basesyntax/service/strategy/PurchaseHandler.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.TypeOfFruit;
+import java.util.List;
+import java.util.Map;
+
+public class PurchaseHandler implements OperationHandler {
+    private final FruitDao dao;
+
+    public PurchaseHandler(FruitDao dao) {
+        this.dao = dao;
+    }
+
+    @Override
+    public void calculateBalanceAfterActivities() {
+
+        List<FruitTransaction> fruiPurchasetList = dao.getInfoFromPurchaseOperationFruit();
+        Map<TypeOfFruit, Integer> balanceMap = dao.getInfoFromBalance();
+
+        for (FruitTransaction fruitInfo : fruiPurchasetList) {
+            balanceMap.replace(fruitInfo.getTypeOfFruit(),
+                    balanceMap.get(fruitInfo.getTypeOfFruit()) - fruitInfo.getQuantity());
+        }
+        dao.rebalanced(balanceMap);
+    }

```

Comment:
Handlers should change storage state according to operation they assigned to

Below example how it may look like:
```suggestion
    @Override
    public void processTransaction(Transaction transaction) {
        int currentQuantity = dao.getAmount(transaction.getFruitName());
        int newQuantity = currentQuantity - transaction.getFruitAmount();
        if (newQuantity < 0) {
            throw new IllegalArgumentException("Purchase cannot be performed. Available stock is less than requested amount)")
        }
        dao.setAmount(newQuantity);
    }

```